contracts/AelinPool.sol

```diff
@@ -385,6 +445,83 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         emit PurchasePoolToken(msg.sender, purchaseTokenAmount);
     }
 
+    /**
+     * @dev allows anyone to become a purchaser with a qualified erc721
+     * nft in the pool depending on the scenarios
+     *
+     * Scenarios:
+     * 1. each wallet holding a qualified NFT to deposit an unlimited amount of purchase tokens
+     * 2. certain amount of purchase tokens per wallet regardless of the number of qualified NFTs held
+     * 3. certain amount of Investment tokens per qualified NFT held
+     */
+
+    function purchasePoolTokensWithNft(
+        address _collectionAddress,
```

Comment:
we should probably let someone pass in an array of arguments here. So like lets say that a pool has 7 collecitons and you have NFTs in each collection. you prob want to pass in an array of addresses and an array of IDS per address, and then the uint256 _purchaseTokenAmount can be passed in separately as a second argument

Reply:
yea that's what i mentioned here https://github.com/AelinXYZ/aelin/pull/59#discussion_r867378507, it'll make the code a little messy, because we won't know which token ids belong to which collection and there can be many token ids per collection. so let's say, an investor has 10 bayc and and 5 mayc, and another has 5 bayc and 5 mayc, we won't know that the first 10 tokenids is for bayc and the next 5 is for mayc for the first investor and first 5 tokenids is for bayc and the next 5 is for mayc for the second investor. 

even if we go with an array of tokenids for a single collection, we won't know how many arrays of tokenids are needed depending on the investor.

Reply:
done. 5e0426fb0e68557fda5f49de0412911a8bbed85b